compile/constant_value.go

```diff
@@ -274,6 +274,7 @@ func (c ConstantSet) Link(scope Scope, t TypeSpec) (ConstantValue, error) {
 	}
 
 	// TODO(abg): Track whether things are linked so that we don't re-link here
+	// TODO(abg): Fail for duplicates
```

Comment:
It's non-trivial to implement because the objects aren't necessarily hashable, and it's low priority enough that it made sense to defer.